### PR TITLE
Add cookbook entry for multilanguage support

### PR DESCRIPTION
**Pull request by erdlet**
Hi all,
I just played around with the internationalization features of the API and thought the results may be interesting for the cookbook section of the homepage. Hope the way I wrote the articles is OK.

Besides, I created an example on how to implement a custom `LocaleResolver` for which I'll write a few lines later.

Looking forward to your feedback :)

### CODE REVIEW

File: src/learn/cookbook/multilang_en.md [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+- Application server
+- Java EE 8
+- MVC API 1.0
+- Eclipse Krazo (for the used application server)
+- Maven
+
+### Project layout
+A simple web application is used for the example. The project layout for this is as follows:
+
+![Projektstruktur](../../img/cookbook/multilang/project_structure.png)
```
Comment (Daniel-Dos) [inline src/learn/cookbook/multilang_en.md]: here is still in DE "Projektstruktur"

Comment (erdlet) [reply src/learn/cookbook/multilang_en.md]: Ah thanks :+1: will be fixed in a few moments

File: src/learn/cookbook/multilang_en.md [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+        </snapshots>
+    </repository>
+</repositories>
+
+<dependencies>
+
+    <dependency>
+      <groupId>javax.mvc</groupId>
+      <artifactId>javax.mvc-api</artifactId>
+      <version>1.0-pfd</version>
```
Comment (chkal) [inline src/learn/cookbook/multilang_en.md]: The static site generator we are using supports to use variables. Therefore, I defined the latest versions of the various artifacts in a configuration file called [versions.json](https://github.com/mvc-spec/www.mvc-spec.org/blob/master/conf/versions.json). So you can use `{{versions.spec.latest}}` here instead which will automatically resolve to `1.0-pfd` at the moment but hopefully to `1.0` soon.

File: src/learn/cookbook/multilang_en.md [hunk 3]
```diff
@@ -0,0 +1,236 @@
...
+    </dependency>
+
+    <!-- Change if your application server doesn't use Jersey -->
+    <dependency>
+      <groupId>org.eclipse.krazo</groupId>
+      <artifactId>krazo-jersey</artifactId>
+      <version>1.0.0-SNAPSHOT</version>
```
Comment (chkal) [inline src/learn/cookbook/multilang_en.md]: Same here, but use `{{versions.krazo.latest}}` this time.

File: src/learn/cookbook/multilang_en.md [hunk 4]
```diff
@@ -0,0 +1,236 @@
...
+
+  /**
+   * Get the assigned message to some key based on the {@link java.util.Locale} of the current request.
+   *
+   * @param key the message key to use
+   * @return the correct translation assigned to the key for the request locale, a fallback translation or
+   * a placeholder for unknown keys.
+   */
+  public final String get(final String key) {
+    final var bundle = ResourceBundle.getBundle(BASE_NAME, mvcContext.getLocale());
```
Comment (chkal) [inline src/learn/cookbook/multilang_en.md]: Just a quick thought about this. My first impression was that maybe the bundle should be only loaded once in a `@PostConstruct` method and the `Messages` bean should be application-scoped instead. However, IIRC `ResourceBundle` provides some caching out of the box, so I'm not sure if my idea is worth the effort. I'm not strong on this one, I just wanted to share my thoughts to hear what others think.

Comment (erdlet) [reply src/learn/cookbook/multilang_en.md]: Initially I thought about an `@ApplicationScope`d Bean with `@PostConstruct` too, but then I won't be able to use multiple `Locale`s, as the `ResourceBundle#getString` doesn't have a parameter for this. And yes, you remember correctly, the `ResourceBundle` is cached out of the box ;)

File: src/learn/cookbook/multilang_en.md (hunk 2 above)
Comment (erdlet) [reply src/learn/cookbook/multilang_en.md]: Ah cool, I'll update this :+1:

File: src/learn/cookbook/multilang_en.md (hunk 4 above)
Comment (chkal) [reply src/learn/cookbook/multilang_en.md]: Oh, yeah. Sorry. You are absolutely correct. My comment didn't make much sense. Sorry. :-)

**Comment by chkal** [comment]
Merged! Thanks a lot!

**Pull request merged**